From 752eafb2d4d3b9035e976a39addc490d6ae02bb3 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@nowhere.nt>
Date: Wed, 21 Oct 2020 16:13:37 +0200
Subject: [PATCH] cmake: linux: Install correct appdata file (#2069).

Reverting to use opencpn.appdata.xml instead of opencpn.metainfo.xml,
the use of metainfo.xml is premature.

Under the hood, flatpak-builder renames opencpn.desktop and
opencpn.appdata.xml to org.opencpn.OpenCPN.* using stanzas in
org.opencpn.OpenCPN.yaml. This ensures the flatpak requirement that
the appdata file is named using the complete, dotted id.

Keeping the names opencpn.appdata.xml and opencpn.desktop fulfills
the Debian requirement that desktop  and appdata file has the same
basename.
---
 CMakeLists.txt              | 2 +-
 ci/generic-build-flatpak.sh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5cf88686e..628e06da6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2228,7 +2228,7 @@ if (NOT APPLE)
     install(
       FILES ${CMAKE_BINARY_DIR}/opencpn.appdata.xml
       DESTINATION ${PREFIX_DATA}/metainfo
-      RENAME opencpn.metainfo.xml
+      RENAME opencpn.appdata.xml
     )
     install(FILES opencpn.1 DESTINATION ${PREFIX_DATA}/man/man1)
   endif (UNIX)
diff --git a/ci/generic-build-flatpak.sh b/ci/generic-build-flatpak.sh
index 6c67a171e..c6fc36c43 100755
--- a/ci/generic-build-flatpak.sh
+++ b/ci/generic-build-flatpak.sh
@@ -75,4 +75,4 @@ flatpak update --appstream opencpn
 flatpak remote-ls opencpn
 
 # Validate the appstream data:
-appstreamcli validate app/files/share/appdata/org.opencpn.OpenCPN.appdata.xml
+appstreamcli validate app/files/share/metainfo/opencpn.appdata.xml
-- 
2.26.2

